packages/react-dom-bindings/src/client/ReactDOMHostConfig.js

```diff
@@ -845,6 +920,284 @@ export function isHydratable(type: string, props: Props): boolean {
+            const rel = element.getAttribute('rel');
+            if (
+              rel === 'stylesheet' &&
+              element.hasAttribute('data-precedence')
+            ) {
+              // This is a stylesheet resource
+              continue;
+            } else if (
+              rel !== anyProps.rel ||
+              element.getAttribute('href') !==
+                (anyProps.href == null ? null : anyProps.href) ||
+              element.getAttribute('crossorigin') !==
+                (anyProps.crossOrigin == null ? null : anyProps.crossOrigin) ||
+              element.getAttribute('title') !==
+                (anyProps.title == null ? null : anyProps.title)
+            ) {
+              // rel + href should usually be enough to uniquely identify a link however crossOrigin can vary for rel preconnect
+              // and title could vary for rel alternate
+              continue;
+            }
+            return element;
+          }
+          case 'style': {
+            if (element.hasAttribute('data-precedence')) {
+              // This is a style resource
+              continue;
+            }
+            // We need to fall through to hit the textContent comparison case below. This switch is structured so that you either
+            // continue, or return, unless you need to do this child comparison.
+            break;
+          }
+          case 'script': {
+            const srcAttr = element.getAttribute('src');
+            if (
+              srcAttr &&
+              element.hasAttribute('async') &&
+              !element.hasAttribute('itemprop')
+            ) {
+              // This is an async script resource
+              continue;
+            } else if (
+              srcAttr !== (anyProps.src == null ? null : anyProps.src) ||
+              element.getAttribute('type') !==
+                (anyProps.type == null ? null : anyProps.type) ||
+              element.getAttribute('crossorigin') !==
+                (anyProps.crossOrigin == null ? null : anyProps.crossOrigin)
+            ) {
+              // This script is for a different src
+              continue;
+            }
+            // We need to fall through to hit the textContent comparison case below. This switch is structured so that you either
+            // continue, or return, unless you need to do this child comparison.
+            break;
+          }
+          default: {
+            // We have excluded the most likely cases of mismatch between hoistable tags, 3rd party script inserted tags,
+            // and browser extension inserted tags. While it is possible this is not the right match it is a decent hueristic
+            // that should work in the vast majority of cases.
+            return element;
+          }
+        }
+
+        // The only branches that should fall through to here are those that need to check textContent against single value children
+        // in particular, <style>, and <script>
```

Comment:
Style isn't reliable. I've found that browsers rewrite those. Not sure about script neither. It's also slow.

I don't think we need to compare this. Just assume it's the one.

Reply:
I can think of a number of ways in which a 3rd party script injecting `<style>` into the body could result in the hydration binding to the wrong instance but they are probably relatively hard to hit. I'll remove this for now and we can see if there are still persistent plugins that get tripped up. We can probably move towards guiding the extension to change than React if there are not very many and their install base is small (the ones that actually get tripped up by this)